include/firmware/common/spi_comms.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <span>
+#include "stm32g4xx_hal_conf.h"
+
+namespace spi {
+    class Spi {
+      public:
+        static constexpr auto BUFFER_SIZE = 5;
+        using BufferType = std::array<uint8_t, BUFFER_SIZE>;
+        Spi();
+        void send_command(BufferType aTxBuffer, uint32_t& data, uint8_t& status);
```

Comment:
This almost certainly wants to either take `aTxBuffer` by reference, e.g. `BufferType& aTxBuffer`, or take a view like a `span`. Defining this as a pass-by-value requires copying the entirety of the buffer every call.